src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,110 @@
+package duke.tasks;
+
+import duke.dukeexceptions.DukeExceptions;
+
+/**
+ * A task class that specific task inherit from.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructor for a task with a description
+     *
+     * @param description the desctiption of the task
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns a string denoting whether the task is done
+     *
+     * @return "X" for task that are done and "" for task that are not.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Sets a task to be done.
+     */
+    public void setDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Sets a task to be not done.
+     */
+    public void setNotDone() {
+        this.isDone = false;
+    }
+
+    /**
+     * retrieves the description of the task.
+     *
+     * @return the description
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * formats the task into a string to be saved locally
+     */
+    public abstract String saveTask();
+
+    /**
+     * decodes a task that has been formatted to be saved locally back into a task.
+     *
+     * @return the task that has been decoded
+     */
+    public static Task decode(String taskString) throws DukeExceptions {
```

Comment:
I've noticed that you have a parser class, how would you make it so that it handles more parsing rather than the Task class handling parsing?